src/bio2bel/sources/biogrid.py

```diff
@@ -0,0 +1,257 @@
+    :param prefix: prefix to filter for (e.g. 'pubmed')
+    :param rstrip: characters to strip from split value from left
+    :param lstrip: characters to strip from split value from right
+    :param list_ids: list of identifiers
+    :return: filtered list of ids
+    """
+    final_list = []
+    for ids in list_ids:
+        id_list = ids.split(separator)
+        flag = False
+        for i in id_list:
+            if i.startswith(prefix):
+                final_list.append(i.lstrip(lstrip).rstrip(rstrip))
+                flag = True
+        if not flag:
+            final_list.append('nan')
+    return final_list
+
+
+def filter_for_prefix_multi(list_ids: Iterable[str], prefix: str, separator: str = '|') -> List[List[str]]:
+    """Split the Iterable by the separator.
+
+    :param separator: separator between ids
+    :param prefix: prefix to filter for (e.g. 'pubmed')
+    :param list_ids: list of identifiers
+    :return: filtered list of lists of ids
+    """
+    final_list = []
+    for ids in list_ids:
+        id_list = ids.split(separator)
+        flag = False
+        row_list = []
+        for i in id_list:
+            if i.startswith(prefix):
+                row_list.append(i)
+                flag = True
+        # if no matching value with prefix is found, append 'nan' as individual value
+        if not flag:
+            final_list.append('nan')
+        # if at least one value is found, append list
+        if len(row_list) > 0:
+            final_list.append(row_list)
+    return final_list
+
+
+def expand_df(df: pd.DataFrame, column_name: str) -> pd.DataFrame:
+    """Add row with same content to dataframe if there are multiple values in one column.
+
+    The specified column must contain an Iterable.
+
+    :param column_name: name of column with multiple values to be simplified
+    :param df: dataframe with multiple values in one column
+    :return: dataframe with expanded values
+    """
+    # identify columns with single values that do not neet to be expanded
+    columns_to_keep = df.columns.to_list()
+    columns_to_keep.remove(column_name)
+
+    # create expanded df
+    lens = [len(item) for item in df[column_name]]
+    df_to_keep = {column: np.repeat(df[column].values, lens) for column in columns_to_keep}
+
+    merged_df = {**df_to_keep, **{column_name: np.concatenate(df[column_name].values)}}
+    return pd.DataFrame(merged_df)
+
+
+def get_processed_biogrid() -> pd.DataFrame:
+    """Load BioGRDID file, filter and rename columns and return a dataframe.
+
+    :return: dataframe of preprocessed BioGRID data
+    """
+    df = _get_my_df()
+    # rename columns
+    df = df.rename(columns=BIOGRID_COLUMN_MAPPER)
+
+    # take relevant columns for source, target, alternative ids, relation and PubMed ID
+    df = df[[SOURCE, TARGET, ALT_SOURCE_ID, ALT_TARGET_ID, RELATION, PUBMED_ID]]
+
+    # filter for uniprot
+    df[SOURCE] = filter_for_prefix_single(list_ids=df[SOURCE], prefix=UNIPROT)
```

Comment:
this is pretty non-obvious. First, since there are several columns for the source node, I again stress to stop using your own constants that are arbitrary wrt the original data. Second, it looks like the identifiers take the form 

```bash
cthoyt@galvac ~/.b/biogrid> cat BIOGRID-ALL-3.5.183.mitab.txt | cut -f 1 | head
#ID Interactor A
entrez gene/locuslink:6416
entrez gene/locuslink:84665
entrez gene/locuslink:90
entrez gene/locuslink:2624
entrez gene/locuslink:6118
entrez gene/locuslink:375
entrez gene/locuslink:377
entrez gene/locuslink:377
entrez gene/locuslink:54464
```
so the question is, what's locuslink, and is the number that follows a proper entrez gene identifier? Because then you can just parse this string very simply with `lambda s: s.split(':')[1]`

Why are you going out of your way to extract the non-preferred identifier?

Reply:
As this is not a Uniprot id, I actually went to the alternative ID column where Uniprot IDs are found. So I filtered that column for the Uniprot IDs and did not take up the Entrez Gene ID.

Reply:
The LocusLink is another identifier. So from my perspective, the id could either be Entrez Gene ID or LocusLink ID. This might also be a problem if you wanted to include the Entrez id.

Reply:
Man, the biogrid website is not forthcoming about details. I just checked a few out and it looks like `entrez gene` and `locuslink` are the same thing (maybe locuslink is some software that they use to index entrez, or resolve its identifiers)

So switch to being simple / more elegant, and use the entrez identifiers. Then, you can use the super secret `xrefs` keyword argument in the `Protein()` class to store the rest. You can use `pyobo.identifier_utils.normalize_curie` to do all of the CURIE parsing too, to make things easier (though you have to install the dev version from https://github.com/pyobo/pyobo.git)

Reply:
See https://github.com/pyobo/pyobo/commit/5bc32225a0d9f69e2591d77277c7aef4caf4b17c. Now that normalize curie function will also recognize these funny prefixes automatically

Reply:
Thanks ;)